pkg/controllers/vdb/init_db.go

```diff
@@ -58,7 +58,9 @@ func (g *GenericDatabaseInitializer) checkAndRunInit(ctx context.Context) (ctrl.
 		return ctrl.Result{}, err
 	}
 
-	if !g.PFacts.doesDBExist() {
+	// create a database if it doesn't exist, or redo the create/revive process if the database creation/revival fails
+	isSet, e := g.Vdb.IsConditionSet(vapi.DBInitialized)
+	if !g.PFacts.doesDBExist() || !isSet || e != nil {
```

Comment:
We aren't checking the error properly. Also, error variables should be called `err` where possible. The status condition check should be enough of a check that we don't need to use `doesDBExist`. There might be some UT pain with removing it, but I think it will be worth it to simplify the check.
```suggestion
	isSet, err := g.Vdb.IsConditionSet(vapi.DBInitialized)
	if err != nil {
	  return ctrl.Result{}, err
	}
	if !isSet {
```